vms/platformvm/service.go

```diff
@@ -1828,6 +1829,48 @@ func (s *Service) GetBlockByHeight(_ *http.Request, args *api.GetBlockByHeightAr
 	return err
 }
 
+// GetFeeConfig returns the dynamic fee config of the chain.
+func (s *Service) GetFeeConfig(_ *http.Request, _ *struct{}, reply *fee.Config) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getFeeConfig"),
+	)
+
+	// TODO: Remove after Etna is activated.
+	now := time.Now()
+	if !s.vm.Config.UpgradeConfig.IsEtnaActivated(now) {
+		return nil
+	}
+
+	*reply = s.vm.DynamicFeeConfig
+	return nil
+}
+
+type GetFeeStateReply struct {
+	fee.State
+	Price fee.GasPrice `json:"price"`
+	Time  time.Time    `json:"timestamp"`
+}
+
+// GetFeeState returns the current fee state of the chain.
+func (s *Service) GetFeeState(_ *http.Request, _ *struct{}, reply *GetFeeStateReply) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getFeeState"),
+	)
+
+	s.vm.ctx.Lock.Lock()
+	defer s.vm.ctx.Lock.Unlock()
+
+	reply.State = s.vm.state.GetFeeState()
+	reply.Price = s.vm.DynamicFeeConfig.MinGasPrice.MulExp(
```

Comment:
Is there shared code that performs this calculation? I'm guessing the price is used somewhere internally, and this is at risk of diverging.

Reply:
I'm not sure how better to share this... This function takes in 3 params to calculate the price... Any helper would also take in 3 params and just call this function... No?